components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts

```diff
@@ -0,0 +1,199 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/* global window */
+
+const { EventEmitter } = require('events')
+import { publicToAddress, toChecksumAddress } from 'ethereumjs-util'
+import {
+  TrezorDerivationPaths, TrezorBridgeAccountsPayload
+} from '../../components/desktop/popup-modals/add-account-modal/hardware-wallet-connect/types'
+import {
+  kTrezorHardwareVendor
+} from '../../constants/types'
+import {
+  TrezorCommand,
+  TrezorFrameCommand,
+  kTrezorBridgeFrameId,
+  kTrezorBridgeUrl,
+  UnlockCommand,
+  GetAccountsCommand,
+  UnlockResponse,
+  GetAccountsResponsePayload,
+  TrezorAccount,
+  postToTrezorFrame,
+  TrezorError
+} from '../../common/trezor/trezor-messages'
+import { getLocale } from '../../../common/locale'
+
+export default class TrezorBridgeKeyring extends EventEmitter {
+  constructor () {
+    super()
+    this.unlocked_ = false
+  }
+
+  type = () => {
+    return kTrezorHardwareVendor
+  }
+
+  getAccounts = async (from: number, to: number, scheme: string) => {
+    if (from < 0) {
+      from = 0
+    }
+    if (!this.isUnlocked() && !(await this.unlock())) {
+      return new Error(getLocale('braveWalletUnlockError'))
+    }
+    return this._getAccounts(from, to, scheme)
+  }
+
+  isUnlocked = () => {
+    return this.unlocked_
+  }
+
+  unlock = () => {
+    return new Promise(async (resolve, reject) => {
+      const messageId = new Date().getTime()
+      this.addEventListener(messageId, (data: UnlockResponse) => {
+        this.unlocked_ = data.result
+        if (data.result) {
+          resolve(true)
+        } else {
+          reject(false)
+        }
+      })
+      const message: UnlockCommand = { id: messageId, origin: window.origin,
+        command: TrezorCommand.Unlock }
+      await this.postMessage(message)
+    })
+  }
+
+  private postMessage = async (command: TrezorFrameCommand) => {
+    let bridge = this.getBridge() as HTMLIFrameElement
+    if (!bridge) {
+      bridge = await this._createBridge() as HTMLIFrameElement
+    }
+    if (!bridge.contentWindow) {
+      throw Error(getLocale('braveWalletCreateBridgeError'))
+    }
+    postToTrezorFrame(bridge.contentWindow, command)
+  }
+
+  private addWindowMessageListener = () => {
+    window.addEventListener('message', this.onMessageReceived.bind(this))
+  }
+
+  private removeWindowMessageListener = () => {
+    window.removeEventListener('message', this.onMessageReceived.bind(this))
```

Comment:
This won't do anything as far as I understand. When you call `Function.bind()` it creates a new function every time. So you're calling to remove event listener that's never been added. However, you don't even need the `.bind(this)` when you add event listener since you're using `=>`  function to preserve `this` scope.

Reply:
done